app/src/main/java/com/github/se/stepquest/services/ScoreServices.kt

```diff
@@ -0,0 +1,112 @@
+package com.github.se.stepquest.services
+
+import com.github.se.stepquest.Friend
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+
+fun getTopLeaderboard(top: Int, callback: (List<Pair<String, Int>>?) -> Unit) {
+  val database = FirebaseDatabase.getInstance()
+  val leaderboardRef = database.reference.child("leaderboard")
+  leaderboardRef
+      .orderByValue()
+      .limitToLast(top)
+      .addListenerForSingleValueEvent(
+          object : ValueEventListener {
+            override fun onDataChange(dataSnapshot: DataSnapshot) {
+              val leaderboard = mutableListOf<Pair<String, Int>>()
+              for (snapshot in dataSnapshot.getChildren()) {
+                val username = snapshot.key ?: continue
+                val score = snapshot.getValue(Int::class.java) ?: continue
+                leaderboard.add(Pair(username, score))
+              }
+              leaderboard.sortByDescending { it.second }
```

Comment:
quick question, is this line necessary when you use leaderboardRef.orderByValue() before?

Reply:
I tried without it and the order wasn't right unfortunately I think it might be because the functions are asynchronous and it stores them in a different order

Reply:
Oh ok I see thanks!